azurelinuxagent/ga/update.py

```diff
@@ -918,3 +958,61 @@ def __str__(self):
             self.last_failure,
             self.failure_count,
             self.was_fatal)
+
+class Supported(object):
+    def __init__(self, path):
+        if path is None:
+            raise UpdateError(u"Supported requires a path")
+        self.path = path
+
+        self._load()
+        return
+
+    @property
+    def is_supported(self):
+        return self._supported_distribution is not None
+
+    @property
+    def in_slice(self):
+        d = self._supported_distribution
+        return d is not None and d.in_slice
+
+    @property
+    def _supported_distribution(self):
+        for d in self.distributions:
+            dd = self.distributions[d]
+            if dd.is_supported:
+                return dd
+        return None
+
+    def _load(self):
+        self.distributions = {}
+        try:
+            if self.path is not None and os.path.isfile(self.path):
+                j = json.loads(fileutil.read_file(self.path))
+                for d in j:
+                    self.distributions[d] = SupportedDistribution(j[d])
+        except Exception as e:
+            logger.warn("Failed JSON parse of {0}: {1}".format(self.path, e))
+        return
+
+class SupportedDistribution(object):
+    def __init__(self, s):
+        if s is None or not isinstance(s, dict):
+            raise UpdateError(u"SupportedDisribution requires a dictionary")
+
+        self.slice = s['slice']
+        self.versions = s['versions']
+
+    @property
+    def is_supported(self):
+        d = ','.join(platform.linux_distribution())
+        for v in self.versions:
+            if re.match(v, d):
+                return True
+        return False
+
+    @property
+    def in_slice(self):
+        n = int((60 * self.slice) / 100)
+        return (n - datetime.utcnow().second) > 0
```

Comment:
@hglkrijger Nope. Seconds range from 0-59 (that is, it is zero-based). Consider the 100% case: n will be 60...which will work for 0-59. :)